groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -116,6 +129,44 @@ public Path getWorkspaceRoot() {
     }
 
     @Override
+    public Map<String, Set<SymbolInformation>> getTypeReferences() {
+        return typeReferences;
+    }
+
+    @Override
+    public Set<SymbolInformation> findReferences(ReferenceParams params) {
+        Set<SymbolInformation> symbols = fileSymbols.get(params.getTextDocument().getUri());
+        if (symbols == null) {
+            return Sets.newHashSet();
+        }
+        List<SymbolInformation> foundSymbolInformations = symbols.stream()
+                .filter(s -> Ranges.isValid(s.getLocation().getRange())
+                            && Ranges.contains(s.getLocation().getRange(), params.getPosition())
+                            && (s.getKind() == SymbolKind.Class || s.getKind() == SymbolKind.Interface
+                                    || s.getKind() == SymbolKind.Enum))
+                // If there is more than one result, we want the symbol whose range starts the latest.
+                .sorted((s1, s2) -> Ranges.POSITION_COMPARATOR.compare(s2.getLocation().getRange().getStart(),
```

Comment:
it's a bit weird that you compare s1 and s2 in the flipped order so that the latest range comes first in the list. Consider using `Comparator.reverseOrder` instead